### PR TITLE
Jwoo/ios container style

### LINKED ISSUE #1929: [iOS] Adaptive Cards not supporting the "emphasis" as container style
Adaptive Cards not supporting the "emphasis" as container style.
RCA :- ACRInputTableView background color is not matching to its parent.

Card JSON:
```json
{
  "body": [
    {
      "spacing": "none",
      "text": "Mohammad Shabaz Moosa created a poll!",
      "type": "TextBlock",
      "wrap": true
    },
    {
      "size": "small",
      "spacing": "none",
      "text": " Votes are visible to everyone.",
      "isSubtle": true,
      "type": "TextBlock",
      "wrap": true
    },
    {
      "items": [
        {
          "size": "extraLarge",
          "wrap": true,
          "color": "accent",
          "type": "TextBlock",
          "text": "Lunch?"
        }
      ],
      "spacing": "extraLarge",
      "type": "Container"
    },
    {
      "style": "emphasis",
      "items": [
        {
          "id": "msteams_vote_option",
          "style": "expanded",
          "type": "Input.ChoiceSet",
          "choices": [
            {
              "title": "Pizza",
              "value": "one"
            },
            {
              "title": "Sushi",
              "value": "two"
            }
          ]
        }
      ],
      "type": "Container",
      "spacing": "extraLarge"
    }
  ],
  "type": "AdaptiveCard",
  "actions": [
    {
      "title": "Submit Vote",
      "data": {
        "post_id": "EwKTA39rsmkEEdW5f",
        "action": "msteams_submit_vote"
      },
      "type": "Action.Submit"
    },
    {
      "title": "Add Comment",
      "card": {
        "type": "AdaptiveCard",
        "actions": [
          {
            "title": "Submit Comment",
            "data": {
              "post_id": "EwKTA39rsmkEEdW5f",
              "action": "msteams_add_comment"
            },
            "type": "Action.Submit"
          }
        ],
        "body": [
          {
            "id": "msteams_add_comment",
            "placeholder": "Add a comment",
            "type": "Input.Text",
            "isMultiline": true
          }
        ]
      },
      "type": "Action.ShowCard"
    }
  ],
  "version": "1.0"
}
```

HOST Config:
```json
{
    "supportsInteractivity": true,
    "imageBaseUrl": "https://pbs.twimg.com/profile_images/3647943215/",
    "fontFamily": "Helvetica Neue",
    "fontSizes": {
        "small": 12,
        "default": 14,
        "medium": 17,
        "large": 21,
        "extraLarge": 26
    },
    "fontWeights": {
        "lighter": 200,
        "default": 400,
        "bolder": 700
    },
    "imageSizes": {
        "small": 40,
        "medium": 80,
        "large": 160
    },
    "containerStyles": {
        "default": {
            "fontColors": {
                "default": {
                    "normal": "#333333",
                    "subtle": "#EE333333"
                },
                "accent": {
                    "normal": "#2E89FC",
                    "subtle": "#882E89FC"
                },
                "good": {
                    "normal": "#54a254",
                    "subtle": "#DD54a254"
                },
                "warning": {
                    "normal": "#c3ab23",
                    "subtle": "#DDc3ab23"
                },
                "attention": {
                    "normal": "#FF0000",
                    "subtle": "#DDFF0000"
                }
            },
            "backgroundColor": "#EAEAEA"
        },
        "emphasis": {
            "fontColors": {
                "default": {
                    "normal": "#333333",
                    "subtle": "#EE333333"
                },
                "accent": {
                    "normal": "#2E89FC",
                    "subtle": "#882E89FC"
                },
                "good": {
                    "normal": "#54a254",
                    "subtle": "#DD54a254"
                },
                "warning": {
                    "normal": "#c3ab23",
                    "subtle": "#DDc3ab23"
                },
                "attention": {
                    "normal": "#FF0000",
                    "subtle": "#DDFF0000"
                }
            },
            "backgroundColor": "#08000000"
              }
    },
    "spacing": {
        "small": 3,
        "default": 8,
        "medium": 20,
        "large": 30,
        "extraLarge": 40,
        "padding": 5
    },
    "separator": {
        "lineThickness": 1,
        "lineColor": "#FF000000"
    },
    "actions": {
        "maxActions": 5,
        "spacing": "Default",
        "buttonSpacing": 10,
        "showCard": {
            "actionMode": "Popup",
            "inlineTopMargin": 16,
            "style": "Emphasis"
        },
        "preExpandSingleShowCardAction": false,
        "actionsOrientation": "Horizontal",
        "actionAlignment": "Stretch"
    },
    "adaptiveCard": {
        "allowCustomStyle": true
    },
    "imageSet": {
        "imageSize": "Medium",
        "maxImageHeight": 999
    },
    "factSet": {
        "title": {
            "size": "Default",
            "color": "Default",
            "isSubtle": false,
            "weight": "Bolder",
            "maxWidth":150,
            "warp": true
        },
        "value": {
            "size": "Default",
            "color": "Default",
            "isSubtle": false,
            "weight": "Default",
            "warp": true
        },
        "spacing": 5
    },
    "media": {
        "playButton" :"https://www.shareicon.net/download/64x64/2017/04/08/883361_media_512x512.png",
        "allowInlinePlayback" : true
    }
}
```

### PR DESCRIPTION
**Pull request by nesalang**
- fixes #1929 
- refactored hex color code conversion

### CODE REVIEW

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm [hunk 1]
```diff
@@ -341,4 +335,61 @@ + (NSArray *)getConstraintsForImageAlignment:(HorizontalAlignment)alignment
     return dateParsedString;
 }
 
++ (UIColor *)convertHexColorCodeToUIColor:(const std::string&)hexColorCode
+{
+    if((hexColorCode.length() < 2) || (hexColorCode.at(0) != '#') || !isxdigit(hexColorCode.at(1)) ||
```
Comment (khouzam) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm]: >isxdigit(hexColorCode.at(1) [](start = 70, length = 27)

Should this validate that all digits are hex?

Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm]: thanks!


---
In reply to: [220330945](https://github.com/Microsoft/AdaptiveCards/pull/1954#discussion_r220330945) [](ancestors = 220330945)

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm [hunk 2]
```diff
@@ -341,4 +335,70 @@ + (NSArray *)getConstraintsForImageAlignment:(HorizontalAlignment)alignment
...
+       ((hexColorCode.length() != 7) && hexColorCode.length() != 9)) {
+        NSLog(@"invalid hexcolor code is given for background color: %@",
```
Comment (bviglietta) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm]: >background color [](start = 51, length = 16)

This method isn't just used for background colors.

Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm]: thanks!
you are absolutley correct.

---
In reply to: [220399342](https://github.com/Microsoft/AdaptiveCards/pull/1954#discussion_r220399342) [](ancestors = 220399342)

Comment (bviglietta) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm]: You should update the log output so it doesn't say "background color".

---
In reply to: [220657960](https://github.com/Microsoft/AdaptiveCards/pull/1954#discussion_r220657960) [](ancestors = 220657960,220399342)

Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm]: I missed that one; i will make the correction. Thanks

---
In reply to: [221090575](https://github.com/Microsoft/AdaptiveCards/pull/1954#discussion_r221090575) [](ancestors = 221090575,220657960,220399342)

**Pull request merged**